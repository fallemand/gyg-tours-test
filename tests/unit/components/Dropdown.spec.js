import React from 'react';
import Dropdown from '../../../app/components/Dropdown';

describe('Dropdown', () => {
  const props = {
    name: '__NAME__',
    title: '__TITLE__',
    onChange: jest.fn(),
    options: [
      { value: '__VAL1__', label: '__LAB1__' },
      { value: '__VAL2__', label: '__LAB2__', selected: true },
      { value: '__VAL1__', label: '__LAB1__', disabled: true },
    ],
  };

  it('render component', () => {
    const component = shallow(<Dropdown {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('expect onChange to be called on changes', () => {
    const component = shallow(<Dropdown {...props} />);
    const event = { target: { value: '__TEST__' } };
    component.instance().handleChange(event);
    expect(props.onChange).toHaveBeenCalledWith('__TEST__');
  });
});
