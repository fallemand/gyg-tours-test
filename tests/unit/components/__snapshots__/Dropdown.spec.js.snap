// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown render component 1`] = `
"<select className='dropdown' name='__NAME__' value='' onChange={[Function: bound handleChange]}>
  <option value='' disabled={true}>
    __TITLE__
  </option>
  <option value='__VAL1__' disabled={[undefined]}>
    __LAB1__
  </option>
  <option value='__VAL2__' disabled={[undefined]}>
    __LAB2__
  </option>
  <option value='__VAL1__' disabled={true}>
    __LAB1__
  </option>
</select>"
`;
